sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -502,6 +502,8 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
         {
           if (renderThreadPrepared || renderEvent.eventType == EventType.DESTROY_RENDERER) {
             renderEvent.runnable?.run()
+          } else if (paused) {
+            nonRenderEventQueue.add(renderEvent)
```

Comment:
Looks safe to me. Side note just for you to consider - we may also change line 498 to be `if (awaitingNextVsync || paused)` - however current solution is perhaps safer although we duplicate code a bit.

Another note - can we actually cover this with robo test - somehow check that without the fix we begin infinite rescheduling while with the fix there's no rescheduling